menas/ui/components/dataset/conformanceRule/upsert.controller.js

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright 2018-2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+sap.ui.define([
+  "sap/ui/core/mvc/Controller"
+], function (Controller) {
+  return Controller.extend("components.dataset.conformanceRule.upsert", {
+
+    rules: [
+      {_t: "CastingConformanceRule"},
+      {_t: "ConcatenationConformanceRule"},
+      {_t: "DropConformanceRule"},
+      {_t: "LiteralConformanceRule"},
+      {_t: "MappingConformanceRule"},
+      {_t: "NegationConformanceRule"},
+      {_t: "SingleColumnConformanceRule"},
+      {_t: "SparkSessionConfConformanceRule"},
+      {_t: "UppercaseConformanceRule"}
+    ],
+
+    dataTypes: [
+      {type: "boolean"},
+      {type: "byte"},
+      {type: "short"},
+      {type: "integer"},
+      {type: "long"},
+      {type: "float"},
+      {type: "double"},
+      {type: "decimal(38,18)"},
+      {type: "char"},
+      {type: "string"},
+      {type: "date"},
+      {type: "timestamp"}
+    ],
+
+    constructor: function () {
+      this._model = sap.ui.getCore().getModel();
+      this._model.setProperty("/rules", this.rules);
+      this._model.setProperty("/dataTypes", this.dataTypes);
+      this._formFragments = {};
+    },
+
+    onBeforeOpen: function () {
+      this._dialog = sap.ui.getCore().byId("upsertConformanceRuleDialog");
+      this._ruleForm = sap.ui.getCore().byId("ruleForm");
+      MappingTableService.getMappingTableList(true, true);
+      if(this._model.getProperty("/newRule/isEdit")) {
+        this.showFormFragment(this._model.getProperty("/newRule/_t"));
+      } else {
+        this._model.setProperty("/newRule/_t", this.rules[0]._t);
+        this.showFormFragment(this.rules[0]._t);
+      }
+      this._dialog.setEscapeHandler(() => this.onClosePress());
+    },
+
+    onAfterOpen: function () {
+      let schemaFieldSelectorSupportedRules =
+        ["CastingConformanceRule", "NegationConformanceRule", "SingleColumnConformanceRule", "UppercaseConformanceRule"];
```

Comment:
The above-mentioned rules could have an attribute `schemaFieldSelectorSupportedRules` which would be false/true. It would be globally accessible in case of future needs.

Reply:
Fixed